core/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -19,23 +19,98 @@
 
 package org.elasticsearch.bootstrap;
 
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import joptsimple.util.KeyValuePair;
+import org.elasticsearch.Build;
+import org.elasticsearch.cli.Command;
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserError;
+import org.elasticsearch.monitor.jvm.JvmInfo;
+
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * This class starts elasticsearch.
  */
-public final class Elasticsearch {
+class Elasticsearch extends Command {
+
+    private final OptionSpec<Void> versionOption;
+    private final OptionSpec<Void> daemonizeOption;
+    private final OptionSpec<String> pathHomeOption;
+    private final OptionSpec<String> pidfileOption;
+    private final OptionSpec<KeyValuePair> propertyOption;
 
     /** no instantiation */
-    private Elasticsearch() {}
+    Elasticsearch() {
+        super("starts elasticsearch");
+        // TODO: in jopt-simple 5.0, make this mutually exclusive with all other options
+        versionOption = parser.acceptsAll(Arrays.asList("V", "version"),
+            "Prints elasticsearch version information and exits");
+        daemonizeOption = parser.acceptsAll(Arrays.asList("d", "daemonize"),
+            "Starts Elasticsearch in the background");
+        // TODO: in jopt-simple 5.0 this option type can be a Path
+        pathHomeOption = parser.acceptsAll(Arrays.asList("H", "path.home"), "").withRequiredArg();
```

Comment:
I have the same concerned I expressed in the previous PR. This should not be a setting a user ever sees or knows about. It is not something they can simply change.